hub/util/transform.py

```diff
@@ -0,0 +1,176 @@
+from hub.util.exceptions import InvalidTransformOutput
+from typing import Any, Callable, Dict, List, Optional, Set
+from hub.util.keys import get_chunk_key, get_index_meta_key, get_tensor_meta_key
+from hub.core.storage.provider import StorageProvider
+from hub.core.meta.index_meta import IndexMeta
+from hub.core.meta.tensor_meta import TensorMeta
+from hub.constants import CHUNK_MAX_SIZE, CHUNK_MIN_TARGET
+
+
+def transform_sample(sample: Any, fn_list: List[Callable], arg_list):
+    """Calls all the functions one after the other on a single sample.
+    Can return 0 or more samples.
+    """
+    result = sample
+    for index in range(len(fn_list)):
+        fn = fn_list[index]
+        kwargs = arg_list[index]
+        if isinstance(result, (list, tuple)) and index != 0:
+            result = [fn(data, **kwargs) for data in result]
+        else:
+            result = fn(result, **kwargs)
+        result = _unwrap(result)
+        verify_transform_output(result)
+    return result if isinstance(result, list) else [result]
+
+
+def _unwrap(ls):
+    """If there is any list then unwrap it into its elements"""
+    items = []
+    for r in ls:
+        if isinstance(r, dict):
+            items.append(r)
+        else:
+            items.extend(r)
+    return items
+
+
+def verify_transform_output(output):
+    if isinstance(output, (list, tuple)):
+        for item in output:
+            if not isinstance(item, dict):
+                print(item)
+                raise InvalidTransformOutput
+    else:
+        if not isinstance(output, dict):
+            raise InvalidTransformOutput
```

Comment:
would it make sense to be grouped into a class so we aren't passing around tuples/dicts with specific fields?